samples/bluetooth/hci_rpmsg/src/main.c

```diff
@@ -447,3 +238,22 @@ void main(void)
 		}
 	}
 }
+
+/* Make sure we register endpoint before IPC Service is initialized. */
+int register_endpoint(const struct device *arg)
+{
+	int status;
+
+	status = rpmsg_service_register_endpoint("nrf_bt_hci", endpoint_cb);
```

Comment:
It would be better to register endpoints using sections: Z_STRUCT_SECTION_ITERABLE. This way, we can remove the runtime dependence which assumes that the endpoint registration must happen before the IPC service library initialization.

We can pass the endpoint id in the callback (Connected event).

Reply:
That's a following improvement that we can address later. I would prefer to keep it out of scope to merge this PR ASAP